subsys/usb/class/hid/core.c

```diff
@@ -526,21 +526,13 @@ static int hid_custom_handle_req(struct usb_setup_packet *setup,
 
 			LOG_DBG("Return HID Descriptor");
 
-			*len = MIN(*len, hid_desc->if0_hid.bLength);
+			*len = MIN(setup->wLength, hid_desc->if0_hid.bLength);
 			*data = (uint8_t *)&hid_desc->if0_hid;
 			break;
 		case USB_DESC_HID_REPORT:
 			LOG_DBG("Return Report Descriptor");
 
-			/* Some buggy system may be pass a larger wLength when
-			 * it try read HID report descriptor, although we had
-			 * already tell it the right descriptor size.
-			 * So truncated wLength if it doesn't match. */
-			if (*len != dev_data->report_size) {
-				LOG_WRN("len %d doesn't match "
-					"Report Descriptor size", *len);
-				*len = MIN(*len, dev_data->report_size);
-			}
```

Comment:
Isn't the `len` parameter here the same as `setup->wLength`? Im not sure but this looks like removing valuable warning message in case len does not match report size previously reported.
Or we simple do not care anymore and thats the reason you removed that? Anyway, just asking, its okay for me.

Reply:
This case can happen, and we should just deal with it. I see no value in the warning.